### ISSUE TITLE
issue to run journey multiple tmes based on user segments. 

### ISSUE THREAD

**Issue by hasnat-venturenox** [dittofeed/dittofeed#958]
We are trying to create a journey on self hosted instance. We have to make sure that the journey should run multiple times per user. For this we have checked the radio button on as shown in picture:
![22](https://github.com/user-attachments/assets/0bfed759-0444-4296-a77f-8f941315efe1)

 

But the issue is that the journey is running only once and when the same event is triggered again the button automatically turns to uncheck and the journey did not run. The journey only run once. 

Steps to reproduces:
1.	Create a segment of users based on properties of event. (i-e I have used 2 properties like “status” and “user_id”.
2.	Create a journey based on the segment as shown in picture. 
 
![33](https://github.com/user-attachments/assets/31d25d53-81e2-48ba-92e4-4874ac650e86)

3.	Mark the flag to run multiple times. 
4.	Start the journey. 

Expected behaviour:
The journey should run when ever the properties meet the condition. For Example, when ever the event occurs in which properties has status draft and user id is 1027. The journey should run. 
Actual Behaviour:
The journey did not run every time and the button which we have checked o run journeys multiple times gets unchecked automatically.

**Comment by maxgurewitz** [comment]
Hi @hasnat-venturenox that's the expected behavior! In order to ensure the user enters the journey multiple times you'll need to do one of the following:

1. Add a time window to your segment to ensure the user can exit, and then re-enter the segment. See the below example where I've also removed the user id condition, which would otherwise restrict the segment to a single user.

<img width="1840" alt="Screen Shot 2024-08-06 at 2 43 49 PM" src="https://github.com/user-attachments/assets/0c56bb56-3756-49eb-bdbf-95389ef5863f">

2. The other option is to create an event entry journey. This will ensure a new journey instance is created with every "assess spec created" event. In contrast, the above segment entry will not run a new concurrent journeys with every event, and can only run at most 1 segment entry journey instance per user at a time.

<img width="1840" alt="Screen Shot 2024-08-06 at 2 45 05 PM" src="https://github.com/user-attachments/assets/a211a4b5-4bf1-4301-a973-acf4a2feecae">

**Comment by maxgurewitz** [comment]
Going to close this for now! Feel free to create another issue or reach out on discord if you have any other questions.
